### PR TITLE
[gbcolor.xml] New software list additions

### PR DESCRIPTION
**Pull request by ICEknigh7**
Just added every known retail release which hadn't been added yet, going by the info in no-intro and other sources. Fixed some bad dumps while leaving the old ones there just in case they could be prototypes, added "submitted" ROM info in many entries which gives us an estimate on the unknown release dates.

These are only those games listed as released, let me know when you want to add all the other GBC stuff (XMLs are ready to copy and paste).

New games:
		AMF Bowling (USA)
		VR Sports Powerboat Racing (USA)

New clones:
		Barcode Taisen Bardigun (Japan, Rev 1)
		Bokujou Monogatari GB3 - Boy Meets Girl (Japan, Rev 1)
		Donkey Kong Country (USA, Not for resale)
		Dragon Quest Monsters 2 - Maruta no Fushigi na Kagi - Ruka no Tabidachi (Japan, Rev 1)
		F1 Championship Season 2000 (Brazil)
		Frogger 2 (USA, Rev 1)
		Fushigi no Dungeon - Fuurai no Shiren GB2 - Sabaku no Majou (Japan, Alt)
		Gakkyuu Ou Yamazaki (Japan)
		Gambler Densetsu Tetsuya - Shinjuku Tenun Hen (Japan, Rev 1)
		Hype - The Time Quest (Brazil)
		Jissen ni Yakudatsu Tsumego (Japan)
		Kawaii Pet Shop Monogatari 2 (Japan, Rev 1)
		Le Roi Lion - Les Adventures de Simba (France, Rev 1)
		Love Hina Pocket (Japan, Rev 1)
		Minna no Shougi - Shokyuu Hen (Japan, Rev 1)
		Mission Impossible (Europe, Rev 1)
		Monster Traveler (Japan)
		Mr. Driller (Japan, NP)
		NBA In the Zone (USA)
		Pocket Puyo Puyo-n (Japan, Rev 1)
		Pocket Puyo Puyo-n (Japan, Rev 2)
		Pokémon - Crystal Version (Aus)
		Pokémon Trading Card Game (Europe, English / Spanish / Italian, Rev 1)
		Pokémon Trading Card Game (Europe, English / French / German, Rev 1)
		Sanrio Timenet - Kako Hen (Japan, Rev 1)
		Sanrio Timenet - Mirai Hen (Japan, Rev 1)
		Shanghai Pocket (Europe)
		Shin Megami Tensei Devil Children - Aka no Sho (Japan, Rev 1)
		Shin Megami Tensei Devil Children - Kuro no Sho (Japan, Rev 1)
		Sewing Machine Operation Software (Europe)
		Super Mario Bros. Deluxe (Japan, NP, Rev 1)
		Tetris Adventure - Susume Mickey to Nakama-tachi (Japan, Rev 1)
		Tom and Jerry - Mousehunt (Europe, Rev 1)
		Trade & Battle Card Hero (Japan, Rev 1)
		Tsuri Sensei 2 (Japan, Rev 1)

Fixed bad dumps:
		The Emperor's New Groove (Europe)
		F1 Racing Championship (Europe)
		Laura (Europe)
		Pro Pool (USA)
		Sei Hai Densetsu (Japan)
		Survival Kids - Kotou no Boukensha (Japan)
		Yakouchuu GB (Japan)

### CODE REVIEW

**Comment by ICEknigh7** [comment]
This would be the rest of it: https://drive.google.com/file/d/1Lov-VczqvXANi1byotQq6uaQcdzojbpN/view?usp=sharing

**Comment by ICEknigh7** [comment]
I've replaced every "Notes: SGB enhanced" comment with a "<feature name="enhancement" value="sgb" />" line, I hope that's okay.

By the way, I've noticed that this list uses the terms "Rev. A, B, etc" just like NES and SNES lists do, but gameboy.xml uses that and "Rev. 1, 2, etc" interchangeably. I don't recall seeing A, B, etc being used on any Nintendo boxes or boards, shall I just change them all to numbers as they appear in no-intro?

**Comment by etabeta78** [comment]
Current values for Revs. in MAME xmls come from either Bootgod's database
(for NES) or no-intro (for SNES and the rest).
There is really no change in the documentation value either way, as long as
it's preserved the order from older revisions to newer ones...

   etabeta

**Comment by ICEknigh7** [comment]
NES, SNES and Game Boy revisions in no-intro all use numbers just like in the boxes/labels/chips, so I'll just use those for consistence with them and with reality.

File: hash/gbcolor.xml [hunk 1]
```diff
@@ -21541,14 +22551,15 @@ Unreleased (music source code exists, possibly no prototypes exist)
...
 	<software name="tsurisn2">
-		<!-- Notes: SGB enhanced -->
 		<description>Tsuri Sensei 2 (Jpn)</description>
 		<year>1999</year>
 		<publisher>J-Wing</publisher>
 		<info name="serial" value="DMG-AF2J-JPN"/>
+		<info name="gold" value="19990624"/>
 		<info name="release" value="19990723"/>
-		<info name="alt_title" value="昆虫博士2"/>
+		<info name="alt_title" value="昆虫博士 ２"/>
```
Comment (cuavas) [inline hash/gbcolor.xml:22560]: Japanese sources use a half-width “2” and no space, e.g. https://w.atwiki.jp/gcmatome/pages/6173.html

File: hash/gbcolor.xml [hunk 2]
```diff
@@ -1289,8 +1329,9 @@ Unreleased (music source code exists, possibly no prototypes exist)
 		<year>1999</year>
 		<publisher>Capcom</publisher>
 		<info name="serial" value="DMG-AUEJ-JPN"/>
+		<info name="gold" value="19991104"/>
 		<info name="release" value="19991224"/>
-		<info name="alt_title" value="爆走戦記 メタルウォーカーGB 鋼鉄の友情"/>
+		<info name="alt_title" value="爆走戦記 メタルウォーカーＧＢ ～鋼鉄の友情～"/>
```
Comment (cuavas) [inline hash/gbcolor.xml:1334]: Do you have a reference for “GB” being styled in fullwidth?

File: hash/gbcolor.xml [hunk 3]
```diff
@@ -6244,14 +6474,15 @@ Unreleased (music source code exists, possibly no prototypes exist)
...
 	<software name="onepyume">
-		<!-- Notes: SGB enhanced -->
 		<description>From TV Animation One Piece - Yume no Luffy Kaizokudan Tanjou! (Jpn, Rev. A)</description>
 		<year>2001</year>
 		<publisher>Banpresto</publisher>
 		<info name="serial" value="DMG-BLUJ-JPN"/>
+		<info name="gold" value="20010615"/>
 		<info name="release" value="20010427"/>
-		<info name="alt_title" value="フロム・テレビ・アニメーション・ワンピース 夢のルフィ海賊団誕生!"/>
+		<info name="alt_title" value="FROM TV ANIMATION　ＯＮＥ ＰＩＥＣＥ　夢のルフィ海賊団誕生！"/>
```
Comment (cuavas) [inline hash/gbcolor.xml:6483]: Do you have a source for “ONE PIECE” being fullwidth?  All the images of it use the regular logo which uses a proportional font.

File: hash/gbcolor.xml [hunk 4]
```diff
@@ -10504,6 +10897,22 @@ Unreleased (music source code exists, possibly no prototypes exist)
 		</part>
 	</software>
 
+	<software name="lionkingfa" cloneof="lionking">
+		<!-- Notes: GBC only -->
+		<!--	In lot check as "cgbblkf1.573"	-->
+		<description>Le Roi Lion - Les Adventures de Simba (Fra, Rev. A)</description>
```
Comment (cuavas) [inline hash/gbcolor.xml:10903]: Does it really have a spelling error in the title?

File: hash/gbcolor.xml [hunk 5]
```diff
@@ -25953,12 +27056,12 @@ These were produced between 2000 and 2001 by Rocket Games, run by Datel Design
...
 	</software>
 
-<!-- This is a title hack of an undumped Sintax game: 真三國無雙 (Zhen San Guo Wu Shuang) whose 'official' english title is True Three Kingdoms -->
 	<software name="cbzz">
-		<!-- Alt. Title: 吞食天地 赤壁之戰 (Devouring of Heaven and Earth: The Battle of Red Cliffs) -->
+	<!-- This is a title hack of an undumped Sintax game: 真三國無雙 (Zhen San Guo Wu Shuang) whose 'official' english title is True Three Kingdoms -->
 		<description>Tun Shi Tian Di - Chi Bi Zhi Zhan (Chi)</description>
 		<year>20??</year>
 		<publisher>&lt;unknown&gt;</publisher>
+		<info name="alt_title" value="吞食天地 赤壁之戰 (Devouring of Heaven and Earth: The Battle of Red Cliffs)"/>
```
Comment (cuavas) [inline hash/gbcolor.xml:27064]: Please don’t put translations in `alt_title` info – they’re too subjective.  Only use official titles.

File: hash/gbcolor.xml (hunk 2 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:1334]: I'm using the official release spreadsheets.

File: hash/gbcolor.xml (hunk 5 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:27064]: I didn't touch any pirate stuff.

Comment (cuavas) [reply hash/gbcolor.xml:27064]: Then why does it show in the diff?

File: hash/gbcolor.xml (hunk 4 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:10903]: That's how it was in the softlist already, I didn't change it.

File: hash/gbcolor.xml (hunk 3 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:6483]: I just copied and pasted it from the official spreadsheets.

File: hash/gbcolor.xml [hunk 6]
```diff
@@ -2827,10 +2916,12 @@ Unreleased (music source code exists, possibly no prototypes exist)
 
 	<software name="carmgedd">
 		<!-- Notes: GBC only -->
+		<!--	USA cartridges have the European sticker underneath.	-->
 		<description>Carmageddon - Carpocalypse Now (Euro, USA)</description>
 		<year>2000</year>
 		<publisher>Titus</publisher>
 		<info name="serial" value="CGB-AKWE-USA, CGB-AKWP-EUR"/>
+		<info name="gold" value="CGB-AKWE-USA, CGB-AKWP-EUR"/>
```
[the remark this answers is not in the document]
Comment (ICEknigh7) [reply hash/gbcolor.xml:2924]: Good catch. Fixed.

File: hash/gbcolor.xml (hunk 1 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:22560]: I just got that from the official spreadsheet, like all the others.

File: hash/gbcolor.xml (hunk 5 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml]: Because those alt names were in the softlist as comments instead of "info" lines.

Comment (ICEknigh7) [reply hash/gbcolor.xml]: I've just removed all those translations.

File: hash/gbcolor.xml (hunk 4 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:10903]: Also, it does indeed: https://youtu.be/ugZlvC8UXfA?t=17

Comment (ICEknigh7) [reply hash/gbcolor.xml:10903]: I've added the (more correct?) title from the box as the "alt_title".

File: hash/gbcolor.xml (hunk 2 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:1334]: I've reverted the change.

File: hash/gbcolor.xml (hunk 3 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:6483]: I've changed it to regular width.

File: hash/gbcolor.xml (hunk 1 above)
Comment (ICEknigh7) [reply hash/gbcolor.xml:22560]: I've reverted the change.

**Comment by ICEknigh7** [comment]
This should be ready to merge now.

**Pull request closed without merging**